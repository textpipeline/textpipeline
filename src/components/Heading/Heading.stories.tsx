import { storiesOf } from '@storybook/react';
import React from 'react';

import Heading, { HeadingProps } from './Heading';

const levels: HeadingProps['level'][] = [1, 2, 3, 4, 5, 6];
const semanticLevels: HeadingProps['semanticLevel'][] = [1, 2, 3, 4, 5, 6];

storiesOf('Heading', module).add('default', () => (
  <>
    {levels.map(level => (
      <Heading key={level} level={level}>
        Heading Level {level}
      </Heading>
    ))}
    {levels.map(level => (
      <>
        {semanticLevels.map(semanticLevel => (
          <Heading key={`${level}-${semanticLevel}`} level={level} semanticLevel={semanticLevel}>
            Heading Level {level} - Semantic Level {semanticLevel}
          </Heading>
        ))}
      </>
    ))}
  </>
));
