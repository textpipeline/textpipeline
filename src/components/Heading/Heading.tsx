import Typography, { TypographyProps } from '@material-ui/core/Typography';
import React from 'react';

type HeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;

export interface HeadingProps {
  readonly level: HeadingLevel;
  readonly color?: TypographyProps['color'];
  readonly align?: TypographyProps['align'];
}

const levelToVisualLevel: Record<HeadingLevel, TypographyProps['variant']> = {
  1: 'h4',
  2: 'h5',
  3: 'h6',
  4: 'subtitle1',
  5: 'body2',
  6: 'body2',
};

const levelToSemanticLevel: Record<HeadingLevel, TypographyProps['component']> = {
  1: 'h1',
  2: 'h2',
  3: 'h3',
  4: 'h4',
  5: 'h5',
  6: 'h6',
};

const Heading: React.FC<HeadingProps> = ({ children, level, align, color = 'secondary' }) => {
  const component = levelToSemanticLevel[level];
  const variant = levelToVisualLevel[level];

  return (
    <Typography variant={variant} component={component} color={color} align={align}>
      <strong>{children}</strong>
    </Typography>
  );
};

export default Heading;
