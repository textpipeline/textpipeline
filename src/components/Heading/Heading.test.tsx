import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Heading, { HeadingProps } from './Heading';

describe('Heading', () => {
  let rendered: ShallowWrapper;

  describe.each([1, 2, 3, 4, 5, 6] as HeadingProps['level'][])('at level %d', (level: HeadingProps['level']) => {
    beforeEach(() => {
      rendered = shallow(<Heading level={level} />);
    });

    it('renders correctly', () => {
      expect(rendered).toMatchSnapshot();
    });

    describe.each([1, 2, 3, 4, 5, 6] as HeadingProps['semanticLevel'][])(
      'with semantic level %d',
      (semanticLevel: HeadingProps['semanticLevel']) => {
        beforeEach(() => {
          rendered = shallow(<Heading level={level} semanticLevel={semanticLevel} />);
        });

        it('renders correctly', () => {
          expect(rendered).toMatchSnapshot();
        });
      }
    );
  });
});
