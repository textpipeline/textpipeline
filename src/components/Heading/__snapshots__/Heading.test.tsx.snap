// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading at level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 1 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h1"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 2 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h2"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 3 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h3"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 4 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h4"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 5 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h5"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 1 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h1"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 2 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h2"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 3 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h3"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 4 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h4"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 5 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h5"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;

exports[`Heading at level 6 with semantic level 6 renders correctly 1`] = `
<WithStyles(ForwardRef(Typography))
  component="h6"
  variant="h6"
>
  <strong />
</WithStyles(ForwardRef(Typography))>
`;
